### PR TITLE
Add retry backoff on client API calls

### PR DESCRIPTION
**Pull request by SiddhantAgarwal**
## CLA

- [ ] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required).
- [ ] Code changes are tested

## Description of the changes, What, Why and How?
Adding a retry interval between two consecutive API calls, the logic is same as in case of web-socket connection retrial added the same kind of behaviour on API calls as well.

### CODE REVIEW

File: src/client.ts [hunk 1]
```diff
@@ -1105,16 +1108,17 @@ export class StreamChat<
...
     } catch (e) {
       this._logApiError(type, url, e);
-
+      this.consecutiveFailures += 1;
       if (e.response) {
         if (
           e.response.data.code === chatCodes.TOKEN_EXPIRED &&
           !this.tokenManager.isStatic()
         ) {
+          await sleep(this._retryInterval());
```
Comment (AnatolyRugalev) [inline src/client.ts:1121]: I think we can retry immediately one time. In the best case scenario token gets refreshed from the first attempt

**Review by AnatolyRugalev — COMMENTED** [review]
Looks good enough :)

File: src/client.ts (hunk 1 above)
Comment (SiddhantAgarwal) [reply src/client.ts:1121]: sure, modifying this.

**Review by AnatolyRugalev — APPROVED** [review]

**Review by ferhatelmas — DISMISSED** [review]
LGTM

File: src/client.ts [hunk 2]
```diff
@@ -1125,6 +1131,13 @@ export class StreamChat<
     }
   };
 
+  _retryInterval = () => {
+    // try to reconnect in 0.25-25 seconds (random to spread out the load from failures)
+    const max = Math.min(500 + this.consecutiveFailures * 2000, 25000);
+    const min = Math.min(Math.max(250, (this.consecutiveFailures - 1) * 2000), 25000);
+    return Math.floor(Math.random() * (max - min) + min);
+  };
```
Comment (mahboubii) [inline src/client.ts:1139]: I think we have a similar implementation for ws class, maybe refactor this func into the util file?

Comment (SiddhantAgarwal) [reply src/client.ts:1139]: Yes it is the same.

Comment (SiddhantAgarwal) [reply src/client.ts]: refactored.

**Review by AnatolyRugalev — APPROVED** [review]

**Review by ferhatelmas — APPROVED** [review]
LGTM

**Review by mahboubii — APPROVED** [review]

**Comment by AnatolyRugalev** [comment]
Something wrong with this PR on Github side of things, it doesn't pick up updates from the branch. Will try to reopen a new one

**Comment by AnatolyRugalev** [comment]
Merged it here: https://github.com/GetStream/stream-chat-js/pull/697

**Pull request closed without merging**